api/base/serializers.py

```diff
@@ -328,6 +350,26 @@ def __init__(self, **kwargs):
         self.link_type = kwargs.pop('link_type', 'url')
         super(JSONAPIHyperlinkedGuidRelatedField, self).__init__(read_only=True, **kwargs)
 
+    def resolve(self, resource):
+        """
+        Resolves the view for target node or target comment when embedding.
+        """
+        embed_value = resource.target._id
+        if resource.target._name == 'node':
```

Comment:
This can be refactored to use a dict defined as a class variable:

class JSONAPIHyperlinkedGuidRelatedField(...):
    # Maps collection name to view and lookup_kwarg
    view_map = {
        'node': {
            'view': 'nodes:node-detail',
            'lookup_kwarg': 'node_id',
        }
    }

    def resolve(self, resource):
        view_info = self.view_map[resource.target._name]
```

Or, even better:

```python
class JSONAPIHyperlinkedGuidRelatedField(...):
    # Maps collection name to view and lookup_kwarg
    view_map = {
        'node': {
            'view': 'nodes:node-detail',
            'view_kwargs': {'node_id': '<_id>'}
        }
    }
```

The latter is more generic and would be able to handle multiple kwargs.